threatfox/package.yaml

```diff
@@ -0,0 +1,151 @@
+
+  threatfox_hashes:
+    type: lookup-table
+    description: The subset of the ThreatFox IOC Database for hashes.
+
+pipelines:
+  fetch-threatfox-ioc-database:
+    name: Update ThreatFox context
+    description: |
+      A pipeline that periodically fetches the ThreatFox IOC database and
+      publishes the feed to the `threatfox` topic.
+    definition: |
+      every {{ inputs.refresh-interval }}
+        from https://threatfox-api.abuse.ch/api/v1/ query=get_iocs days:={{ inputs.days }}
+      | yield data[]
+      // This is a hack that we need because there are no union types. We
+      // roundtrip via the JSON parser which does per-event type inference for
+      // us.
+      | write json -c
+      | read json --ndjson --precise
+      | publish threatfox
+
+  # We are currently creating one pipeline per IOC type. Once we have a more
+  # powerful group operator, it would be nice to write some thing like
+  # `group ioc_type { ... }` and then update the corresponding context
+  # dynamically.
+  populate-context-ipports:
+    name: Update the ThreatFox context for (IP, port) tuples
+    description: |
+      A pipeline that puts (IP, port) tuples into the corresponding context.
+    definition: |
+      // tql2
+      | subscribe "threatfox"
+      | where ioc_type == "ip:port"
+      | ioc = ioc.grok("%{IP:ip}:%{NUMBER:port}")
+      | legacy "context update threatfox_ipports --key=ioc"
+
+  populate-context-domains:
+    name: Update the ThreatFox context for domains
+    description: |
+      A pipeline that puts domains into the corresponding context.
+    definition: |
+      // tql2
+      | subscribe "threatfox"
+      | where ioc_type == "domain"
+      | legacy "context update threatfox_domains --key=ioc"
+
+  populate-context-urls:
+    name: Update the ThreatFox context for domains
+    description: |
+      A pipeline that puts domains into the corresponding context.
+    definition: |
+      // tql2
+      | subscribe "threatfox"
+      | where ioc_type == "url"
+      | legacy "context update threatfox_domains --key=ioc"
+
+  populate-context-hashes:
+    name: Update the ThreatFox context for domains
+    description: |
+      A pipeline that puts domains into the corresponding context.
+    definition: |
+      // tql2
+      | subscribe "threatfox"
+      | where ioc_type.ends_with("hash")
+      | legacy "context update threatfox_hashes --key=ioc"
+
+snippets:
+  - name: Enrich Zeek DNS logs with ThreatFox domains
+    description: |
+      Enriches Zeek DNS logs with the ThreatFox IOCs of type `domain`.
+    definition: |
+      subscribe zeek
+      | where #schema == "zeek.dns"
+      | enrich threatfox_domains --field=query
+
+  - name: ThreatFox IOCs by confidence
+    description: |
+      Shows a bar chart that plots the number of IOCs per confidence level.
+    definition: |
+      from https://threatfox-api.abuse.ch/api/v1/ query=get_iocs days:={{ inputs.days }}
```

Comment:
Can't easily test this in the app because it's not merged yet, but I'm not sure if we support inputs in examples.